Sources/MessagingInApp/Gist/Utilities/UIColor+Hex.swift

```diff
@@ -0,0 +1,35 @@
+import UIKit
+
+extension UIColor {
+    static func fromHex(color: String?) -> UIColor? {
+        guard var hexColor = color else { return nil }
+        if hexColor.hasPrefix("#") {
+            hexColor.removeFirst()
+        }
+
+        let scanner = Scanner(string: hexColor)
+
+        var rgbValue: UInt64 = 0
+        guard scanner.scanHexInt64(&rgbValue) else {
+            return nil
+        }
+
+        var red, green, blue, alpha: UInt64
+        switch hexColor.count {
+        case 6:
+            red = (rgbValue >> 16)
+            green = (rgbValue >> 8 & 0xFF)
+            blue = (rgbValue & 0xFF)
+            alpha = 255
+        case 8:
+            red = (rgbValue >> 16)
+            green = (rgbValue >> 8 & 0xFF)
+            blue = (rgbValue & 0xFF)
+            alpha = rgbValue >> 24
+        default:
+            return nil
+        }
+
+        return UIColor(red: CGFloat(red) / 255, green: CGFloat(green) / 255, blue: CGFloat(blue) / 255, alpha: CGFloat(alpha) / 255)
+    }
```

Comment:
```suggestion
 static func fromHex(_ hex: String?) -> UIColor? {
        guard let hex = hex else { return nil }
        
        let cleanHex = hex.hasPrefix("#") ? String(hex.dropFirst()) : hex
        
        // Validate hex string length
        guard cleanHex.count == 6 || cleanHex.count == 8 else {
            return nil
        }
        
        var rgbValue: UInt64 = 0
        guard Scanner(string: cleanHex).scanHexInt64(&rgbValue) else {
            return nil
        }
        
        // Extract color components
        let red = CGFloat((rgbValue >> 16) & 0xFF) / 255.0
        let green = CGFloat((rgbValue >> 8) & 0xFF) / 255.0
        let blue = CGFloat(rgbValue & 0xFF) / 255.0
        let alpha = cleanHex.count == 8 
            ? CGFloat((rgbValue >> 24) & 0xFF) / 255.0 
            : 1.0
        
        return UIColor(red: red, green: green, blue: blue, alpha: alpha)
    }
```

small refactor and optimizations